OrbitLinuxTracing/TracerThread.cpp

```diff
@@ -52,28 +54,29 @@ void TracerThread::Run(
                                                BIG_RING_BUFFER_SIZE_KB};
         if (uprobe_ring_buffer.IsOpen()) {
           fds_to_ring_buffer.emplace(uprobe_fd, std::move(uprobe_ring_buffer));
-          uprobe_fds_to_function.emplace(uprobe_fd, &function);
+          uprobe_fds_to_function_.emplace(uprobe_fd, &function);
         }
 
         int uretprobe_fd = uretprobes_event_open(
             function.BinaryPath().c_str(), function.FileOffset(), -1, cpu);
-        PerfEventRingBuffer uretprobe_ring_buffer{uretprobe_fd,
-                                                  SMALL_RING_BUFFER_SIZE_KB};
-        if (uretprobe_ring_buffer.IsOpen()) {
-          fds_to_ring_buffer.emplace(uretprobe_fd,
-                                     std::move(uretprobe_ring_buffer));
-          uretprobe_fds_to_function.emplace(uretprobe_fd, &function);
-        }
+        uret_probes_fds.push_back(uretprobe_fd);
+
+        // Redirect uretprobes to uprobe ring buffer to reduce number of ring
+        // buffers and to coalesce closely related events.
+        int ret = ioctl(uretprobe_fd, PERF_EVENT_IOC_SET_OUTPUT, uprobe_fd);
+        if (ret) ERROR("PERF_EVENT_IOC_SET_OUTPUT error: %i.\n", ret);
+        ret =
+            ioctl(uretprobe_fd,
+                  PERF_EVENT_IOC_ENABLE);  // TODO: replace by perf_event_enable
```

Comment:
Given the TODO, I assume that ioctl is called directly here to check the return value? Can we check it and in case print the error directly in perf_event_enable (and do the same in perf_event_disable)?